firehose/firehose.go

```diff
@@ -0,0 +1,242 @@
+package firehose
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/firehose"
+	"github.com/aws/aws-sdk-go-v2/service/firehose/types"
+)
+
+const (
+	region        = "AWS_REGION"
+	defaultRegion = "us-east-1"
+)
+
+//Config aws configuration
+type Config struct {
+	Endpoint string
+	Region   string
+}
+
+// PutResponse per record
+type PutResponse struct {
+	RecordID string
+	Error    error
+}
+
+// ClientProvider for kinesis firehose
+type ClientProvider struct {
+	firehose *firehose.Client
+	region   string
+	endPoint string
+}
+
+// NewClientProvider initiate new client provider
+func NewClientProvider() (*ClientProvider, error) {
+	c := &ClientProvider{}
+	c.region = os.Getenv(region)
```

Comment:
AFAIK, we need to use `viper` for environment variables and if env value is not set, we may fetch from AWS SSM.
Below are the few references from `da-trends-caching` repo -
https://github.com/LF-Engineering/dev-analytics-trends-caching/blob/master/.trends-caching.yaml
https://github.com/LF-Engineering/dev-analytics-trends-caching/blob/master/util/parameters.go#L11

CC: @fayazg

Reply:
It will be a config package to take care of this, it's not this PR scope